### PR TITLE
Record and report last_frame_alive for each bot

### PR DESCRIPTION
**Pull request by erdman**
in both verbose and quiet.  Last_frame_alive reports how far into the game each bot survives ... more color than just rank finish.  Also, it's the score for single player mode.

### CODE REVIEW

File: environment/main.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ int main(int argc, char ** argv) {
         std::cout << stats;
     }
     else {
-        for(unsigned int a = 0; a < stats.player_statistics.size(); a++) std::cout << "Player #" << stats.player_statistics[a].tag << ", " << my_game->getName(stats.player_statistics[a].tag) << ", came in rank #" << stats.player_statistics[a].rank << "!\n";
+        for(unsigned int a = 0; a < stats.player_statistics.size(); a++) std::cout << "Player #" << stats.player_statistics[a].tag << ", " << my_game->getName(stats.player_statistics[a].tag) << ", came in rank #" << stats.player_statistics[a].rank << ", and was last alive on frame #" << stats.player_statistics[a].last_frame_alive << "!\n";
```
Comment (Janzert) [inline environment/main.cpp]: That's known as an Oxford comma (or sometimes Serial comma) and is optional depending in the style guide you are using. American style guides more commonly require it and British ones less commonly (The Oxford Style Manual as the comma name suggests being one of the minority British guides).

In other words it is correct with or without it, unless you need to follow a specific style that requires one way or the other.

**Comment by truell20** [comment]
I will have to edit the environment command reference and our worker script before merging.

**Comment by erdman** [comment]
Just realized this needs to be tested when games go to the frame limit and multiple players make it to the last frame.  I expect an off by one error in that case.  I'll get back to you.

**Comment by erdman** [comment]
Off-by-one error in case of ties at maxTurnNumber did exist ... pushed another commit here to fix.

File: environment/main.cpp (hunk 1 above)
Comment (truell20) [inline environment/main.cpp]: Wdyt @Janzert?

**Pull request merged**